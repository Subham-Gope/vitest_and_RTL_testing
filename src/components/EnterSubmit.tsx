import React, { useState } from "react";
import type { ChangeEvent, KeyboardEvent } from "react";

interface PropType {
  message: string;
}

const EnterSubmit: React.FC<PropType> = ({ message }) => {
  const [value, setValue] = useState<string>("");
  const [submitted, setSubmitted] = useState<string>("");

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    setValue(e.target.value);
  };

  const handleKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
    if (e.key == "Enter") {
      e.preventDefault();
      setSubmitted(value);
    }
  };

  return (
    <>
      <h1>{message}</h1>
      <div>
        <label htmlFor="enter-submit">Enter Your Text</label>
        <input
          type="text"
          id="enter-submit"
          placeholder="Enter your text"
          value={value}
          onChange={handleChange}
          onKeyDown={handleKeyDown}
        />
      </div>

      {submitted && <p>Submitted: {submitted}</p>}
    </>
  );
};

export default EnterSubmit;
