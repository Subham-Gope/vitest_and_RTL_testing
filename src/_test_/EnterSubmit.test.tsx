import { describe, test, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import EnterSubmit from "../components/EnterSubmit";

describe("Testing the EnterSubmit.tsx component", () => {
  test("submits input on ENter key press", async () => {
    // setting up userEvent
    const user = userEvent.setup();

    const { container } = render(
      <EnterSubmit message={"Hello there we are learning component testing"} />
    );
    expect(container).toMatchSnapshot();

    const input = screen.getByRole("textbox", { name: /enter your/i });
    expect(input).toBeInTheDocument();

    // intiating user action
    await user.type(input, "Hello there Sumon Barcha");
    await user.keyboard("{Enter}");

    const summaryText = screen.getByText(/Submitted:/i);
    expect(summaryText).toBeInTheDocument();
    expect(summaryText).toHaveTextContent("Hello there Sumon Barcha");
  });
});
