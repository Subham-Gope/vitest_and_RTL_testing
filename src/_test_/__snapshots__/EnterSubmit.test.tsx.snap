// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the EnterSubmit.tsx component > submits input on ENter key press 1`] = `
<div>
  <h1>
    Hello there we are learning component testing
  </h1>
  <div>
    <label
      for="enter-submit"
    >
      Enter Your Text
    </label>
    <input
      id="enter-submit"
      placeholder="Enter your text"
      type="text"
      value=""
    />
  </div>
</div>
`;
